src/lib/Syntax.hs

```diff
@@ -137,7 +139,10 @@ data Atom (n::S) =
  | ACase (Atom n) [AltP Atom n] (Type n)
    -- single-constructor only for now
  | DataConRef (DataDefName n) [Atom n] (EmptyAbs (Nest DataConRefBinding) n)
- | BoxedRef (Atom n) (Block n) (Abs Binder Atom n)  -- ptr, size, binder/body
+ -- lhs ref, rhs ref abstracted over the eventual value of lhs ref, type
+ | DepPairRef (Atom n) (Abs Binder Atom n) (DepPairType n)
+ | BoxedRef [(Atom n, Block n)]        -- ptrptrs/sizes
```

Comment:
Can you add a note explaining when can the block be evaluated? I assume it references memory that starts uninitialized and can only be safely used after some condition is satisfied.

Reply:
The block can be evaluated any time (but it usually lives under an abstraction, such at the one in `DepPairRef`. The atom is a pointer, with the usual initialization story.